src/named_tuple.cr

```diff
@@ -582,7 +586,7 @@ struct NamedTuple
     {% end %}
   end
 
-  private def first_key_internal
+  private def first_key_internal : Symbol
```

Comment:
This does not have the same effect as you might think because `NoReturn < Symbol` and is therefore still a valid return type.

Reply:
I noticed that change had no effect, so I manually added the casts to `Symbol`

I'll remove the annotation 👍

Reply:
I don't think `as(Symbol)` has a useful effect either, if `self` is empty then the block will simply be untyped.

Reply:
Seems to pass the specs I wrote fine

https://github.com/crystal-lang/crystal/pull/10942/files#diff-6cb2ca5f2ef59972215914e5b34ce7203f07d98c64babb4bb68f5bad6c0ddb79R350